pylint/checkers/unicode.py

```diff
@@ -0,0 +1,537 @@
+            if pos != ignore_pos:
+                # Calculate the column
+                col = int(pos / byte_str_length)
+                result[col] = char
+            start = pos + 1
+            pos = line.find(search_for, start)
+
+    return result
+
+
+UNICODE_BOMS = {
+    "utf-8": codecs.BOM_UTF8,
+    "utf-16": codecs.BOM_UTF16,
+    "utf-32": codecs.BOM_UTF32,
+    "utf-16le": codecs.BOM_UTF16_LE,
+    "utf-16be": codecs.BOM_UTF16_BE,
+    "utf-32le": codecs.BOM_UTF32_LE,
+    "utf-32be": codecs.BOM_UTF32_BE,
+}
+BOM_SORTED_TO_CODEC = OrderedDict(
+    # Sorted by length of BOM giving the codec
+    (UNICODE_BOMS[codec], codec)
+    for codec in ("utf-32le", "utf-32be", "utf-8", "utf-16le", "utf-16be")
+)
+
+UTF_NAME_REGEX_COMPILED = re.compile(
+    "utf[ -]?(8|16|32)[ -]?(le|be|)?(sig)?", flags=re.IGNORECASE
+)
+
+
+def _normalize_codec_name(codec: str) -> str:
+    """Make sure the codec name is always given as defined in the BOM dict"""
+    return UTF_NAME_REGEX_COMPILED.sub(r"utf-\1\2", codec).lower()
+
+
+def _remove_bom(encoded: bytes, encoding: str) -> bytes:
+    """remove the bom if given from a line"""
+    if not encoding.startswith("utf"):
+        return encoded
+    bom = UNICODE_BOMS[encoding]
+    if encoded.startswith(bom):
+        return encoded[len(bom) :]
+    return encoded
+
+
+def _encode_without_bom(string: str, encoding: str) -> bytes:
+    """Encode a string but remove the BOM"""
+    return _remove_bom(string.encode(encoding), encoding)
+
+
+def _byte_to_str_length(codec: str) -> int:
+    """return how many byte are usually(!) a character point"""
+    if codec.startswith("utf-32"):
+        return 4
+    if codec.startswith("utf-16"):
+        return 2
+
+    return 1
+
+
+@lru_cache(maxsize=1000)
+def _cached_encode_search(string: str, encoding: str) -> bytes:
+    """A cached version of encode used for search pattern"""
+    return _encode_without_bom(string, encoding)
+
+
+def _fix_utf16_32_line_stream(steam: Iterable[bytes], codec: str) -> Iterable[bytes]:
+    """Handle line ending for UTF16 and UTF32 correctly
+
+    Currently Python simply strips the required zeros after \n after the
+    line ending. Leading to lines that can't be decoded propery
+    """
+    if not codec.startswith("utf-16") and not codec.startswith("utf-32"):
+        yield from steam
+    else:
+        # UTF 16 and UTF-32 are sucking, so first get all the bytes in memory
+        content = b"".join(line for line in steam)
+
+        new_line = _cached_encode_search("\n", codec)
+
+        # Now we split the line by the real new line in the correct encoding
+        # we can't use split as it would strip the \n that we need
+        start = 0
+        while True:
+            pos = content.find(new_line, start)
+            if pos >= 0:
+                yield content[start : pos + len(new_line)]
+            else:
+                # Yield the rest and finish
+                yield content[start:]
+                break
```

Comment:
```suggestion
```

I believe this is unnecessary. If not we should probably write a test for it.

Reply:
How can a break a in endless loop not be required, if no "EndIteration" is reaised? 
I adopted the test to not use zip therefore hit the brake.
Doing so I even realized a small bug within the unittest/function.

Reply:
Well, if the endless loop is set up in such a way that either of the conditions will always be reached it is unnecessary.
I mean, that's what was happening with the previous tests. Thanks for changing the test though 😄